### PR TITLE
[CIS GCP] Implement compute rules: 4.8, 4.9, 4.11

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
Implement rules 4.8, 4.9, 4.11

### Related Issues
- Related: https://github.com/elastic/csp-security-policies/issues/241

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_confidential_computing_enabled),
+		{"Compute instance": input.resource},
+	)
+}
+
+is_confidential_computing_enabled {
+	data_adapter.resource.data.confidentialInstanceConfig
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego:24]: is this line redundant?

Comment (oren-zohar) [reply bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego:24]: if so we can do `is_confidential_computing_enabled := data_adapter.resource.data.confidentialInstanceConfig.enableConfidentialCompute` instead?

File: bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	data_adapter.resource.data.confidentialInstanceConfig.enableConfidentialCompute
+}
+
+get_machine_type_family(type_url) = family {
+	parts := split(type_url, "/")
+	family := parts[count(parts) - 1]
+}
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego:31]: can you please move to common and add the format assumption in the documentation for this parsing?

File: bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package compliance.cis_gcp.rules.cis_4_8
+
+import data.compliance.lib.common
+import data.compliance.policy.gcp.data_adapter
+
+default is_shielded_vm = false
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego:6]: any reason to use defailt ... = false instead of else = false clause in the audit?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego (hunk 3 above)
Comment (uri-weisman) [reply bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego:6]: no reason, converted.

File: bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego (hunk 2 above)
Comment (uri-weisman) [reply bundle/compliance/cis_gcp/rules/cis_4_11/rule.rego:31]: moved to gcp's common

File: bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego (hunk 3 above)
Comment (oren-zohar) [reply bundle/compliance/cis_gcp/rules/cis_4_8/rule.rego]: please convert the other rules too

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**